scirpy/datasets/__init__.py

```diff
@@ -205,3 +206,138 @@ def vdjdb(cached: bool = True, *, cache_path="data/vdjdb.h5ad") -> AnnData:
     adata.write_h5ad(cache_path)
 
     return adata
+
+
+def iedb(cached: bool = True, *, cache_path="data/iedb.h5ad") -> AnnData:
+    """\
+    Download IEBD v3 and process it into an AnnData object.
+
+    :cite:`iedb` is a curated database of
+    T-cell receptor (TCR) sequences with known antigen specificities.
+
+    Parameters
+    ----------
+    cached
+        If `True`, attempt to read from the `data` directory before downloading
+
+    Returns
+    -------
+    An anndata object containing all entries from IEDB in `obs`.
+    Each entry is represented as if it was a cell, but without gene expression.
+    Metadata is stored in `adata.uns["DB"]`.
+    """
+    if cached:
+        try:
+            return sc.read_h5ad(cache_path)
+        except OSError:
+            pass
+
+    logger = _IOLogger()
+    iedb_df = pd.read_csv(
+        "https://www.iedb.org/downloader.php?file_name=doc/receptor_full_v3.zip",
+        index_col=None,
+        sep=",",
+        na_values=["None"],
+        true_values=["True"],
+        low_memory=False,
+    )
+
+    iedb_df = iedb_df.drop_duplicates()
+    iedb_df = iedb_df.drop_duplicates(
+        [
+            "Chain 1 CDR3 Curated",
+            "Chain 2 CDR3 Curated",
+            "Organism",
+            "Antigen",
+            "Response Type",
+            "Chain 1 Type",
+            "Chain 2 Type",
+        ]
+    )
+
+    iedb_df.loc[
+        iedb_df["Chain 1 CDR3 Curated"].isna(), "Chain 1 CDR3 Curated"
+    ] = iedb_df["Chain 1 CDR3 Calculated"]
+
+    iedb_df.loc[
+        iedb_df["Chain 2 CDR3 Curated"].isna(), "Chain 2 CDR3 Curated"
+    ] = iedb_df["Chain 2 CDR3 Calculated"]
+
+    iedb_df["Chain 1 CDR3 Curated"] = iedb_df["Chain 1 CDR3 Curated"].str.upper()
+    iedb_df["Chain 2 CDR3 Curated"] = iedb_df["Chain 2 CDR3 Curated"].str.upper()
```

Comment:
One more request: Do you think it would make sense to also merge the Calculated/Computed for the V/D/J gene names?

Reply:
Thank you for fixing the light chain issue!
Yes, makes sense also to replace the V/D/J gene names if not present with the calculated ones, this is included in my new commit, let me know what you think.